symmer/symplectic/base.py

```diff
@@ -999,6 +998,64 @@ def to_sparse_matrix(self):
         )
         return sparse_Qstate
 
+    def sample_state(self, n_samples) -> dict:
+        """
+        Method to sample given quantum state in computational basis. Get an array of bitstrings and counts as output.
+
+        #TODO: could have change of operator added prior to sample
+
+        Args:
+            n_samples: how many bitstring samples to take
+
+        Returns:
+            self.state_matrix: matrix, where rows represent bitstrings
+            counter: number of times each row was sampled
+        """
+        counter = np.random.multinomial(n_samples, np.abs(self.state_op.coeff_vec)**2)
```

Comment:
fast way to sample a state!

qiskit uses: https://github.com/Qiskit/qiskit-terra/blob/6abb8824e54ff7fccb589fe6a0f3912ebebd95d0/qiskit/quantum_info/states/quantum_state.py#L271

np unique is SLOW!

Reply:
@TimWeaving look at commented out part above return statement... We could return as a qubit state, with coeffs = number of samples! What do you think?

Reply:
Agreed on returning as QuantumState, with option to return raw counts or normalized state